agent/local/state_test.go

```diff
@@ -1606,6 +1605,59 @@ func TestAgent_AddCheckFailure(t *testing.T) {
 	}
 }
 
+func TestAgent_AliasCheck(t *testing.T) {
+	t.Parallel()
+
+	require := require.New(t)
+	cfg := config.DefaultRuntimeConfig(`bind_addr = "127.0.0.1" data_dir = "dummy"`)
+	l := local.NewState(agent.LocalConfig(cfg), nil, new(token.Store))
+	l.TriggerSyncChanges = func() {}
+
+	// Add checks
+	require.NoError(l.AddService(&structs.NodeService{Service: "s1"}, ""))
+	require.NoError(l.AddService(&structs.NodeService{Service: "s2"}, ""))
+	require.NoError(l.AddCheck(&structs.HealthCheck{CheckID: types.CheckID("c1"), ServiceID: "s1"}, ""))
+	require.NoError(l.AddCheck(&structs.HealthCheck{CheckID: types.CheckID("c2"), ServiceID: "s2"}, ""))
+
+	// Add an alias
+	notifyCh := make(chan struct{}, 1)
+	require.NoError(l.AddAliasCheck(types.CheckID("a1"), "s1", notifyCh))
+
+	// Update and verify we get notified
+	l.UpdateCheck(types.CheckID("c1"), api.HealthCritical, "")
+	select {
+	case <-notifyCh:
+	default:
+		t.Fatal("notify not received")
+	}
+
+	// Update again and verify we do not get notified
+	l.UpdateCheck(types.CheckID("c1"), api.HealthCritical, "")
+	select {
+	case <-notifyCh:
+		t.Fatal("notify received")
+
+	case <-time.After(50 * time.Millisecond):
```

Comment:
change to `default:`

Reply:
I thought these would be better as a short timer to help check we dont receive a channel send (ie if the code was changed to async by accident). 50ms is not too long so maybe it’s not even worth it but from a test perspective we want to check we don’t receive a channel send.

Behaviorally TODAY this could be default but was wondering if from a test perspective it’s worth loosening to try to catch issues? Or should we just do default?

Reply:
Putting the time.After here I think could provide a false sense of security. If we did change the function to be async its likely that 50 or 100ms wait times would not be enough (especially when heavily parallelized or running in environments like travis). For these cases where we are checking that we dont receive a notification its likely it would always pass (even when it shouldn't). For the checks where we are expecting a notification a 50 - 100ms wait could be to little and we would have periodic test failures. The only way to make this test bullet-proof would be to have an exceedingly long timeout but the disadvantage is that it would make running this test take much longer. Even when we do not currently need it to be.

My preference is to make the tests as accurate and fast as possible to reduce the feedback loop time during development. In this case making the assumption that the underlying function is not async seems reasonable. Also if the function does become async, the first check with expecting a notification should fail quickly and it would immediately identify the need to change/rewrite the test to accommodate it being async.

Reply:
Sounds good. I’ll make the change when I get back to a computer!

Reply:
Done!